### ISSUE TITLE
No default values for doi or version in load_austraits

### ISSUE THREAD

**Issue by fontikar** [traitecoevo/austraits#34]
**Describe the bug**
This pertains to `load_austraits()` and was brought to my attention by @wcornwell! Thanks Will! :) 

Currently the arguments versions and doi are set to NULL. The reason behind this decision is because the user can either supply a `version` or a `doi`.  Traditionally, if users see this, it implies those arguments are optional and not necessary to set. This is a source of confusion because our intention is for users to supply this and this is usually achieved by not setting a default value in the function. 

Within the `load_austraits()` function there are some `if` statements that flags that if `doi` is not supplied , it will match its respective `version` number and then proceed to check if the `version` or `doi` exist; then retrieve it. (See attached screenshot)  Currently, `load_austraits()`operates through knowing the version number, but this can certainly refined!  

![Screen Shot 2022-03-08 at 2 03 16 pm](https://user-images.githubusercontent.com/12161802/157158160-c37bb1ea-a269-4b00-9c0a-9fb319d296c6.png)

**Expected behavior**
I tried removing `= NULL` for `doi` and `version` but this trips my existing `if` statements because the function is expecting BOTH `version` and `doi` to be supplied as oppose to **one or the other**. 

Is there a way to improve this function and code this, ideally without `= NULL` for `doi` and `version`

**Comment by wcornwell** [comment]
I think the R-ish way to do it would be to have one (required) user input which is then internally parsed into either a doi or a version number.  Then the standard R system for missing required inputs would work as designed.  

There are other solutions, but none of them will fit what R users are expecting...

**Comment by fontikar** [comment]
In [2d1a3b9](https://github.com/traitecoevo/austraits/commit/2d1a3b9a7be2c9970f0c74142b27426f4d4e3856) I have added `version = "3.0.2"` as a default. Also I built in the function that if doi and version is both supplied and don't match up, the version will be taken. This gives the default version precedence and users will need to specific version = NULL if they choose to use doi...need to brain storm a bit! Either no version default or the function just stops if version and doi doesn't match. doi does seem redundant at this point...
